augur/export_v2.py

```diff
@@ -16,6 +17,10 @@
 from .utils import read_node_data, write_json, read_config, read_lat_longs, read_colors
 from .validate import export_v2 as validate_v2, auspice_config_v2 as validate_auspice_config_v2, ValidateError
 
+
+MINIFY_THRESHOLD = 1000 # number of nodes
```

Comment:
I'm thinking this should be lower. @jameshadfield just sent me an example of a dataset with 1176 nodes which was 52MB un-minified, with `node_attrs` and all, and took 17 seconds to load for me. This means a dataset with 999 nodes would result in a similar experience. Would a threshold of 100 be unreasonable?

Reply:
I don't know if that's the best example, or perhaps it's a perfect example!

The reason that dataset is so large in size is because it contains thousands of mutations (it's a 4Mb genome). Those mutations are encoded with new-lines between every. single. mutation. so it would be a perfect candidate for minimisation. Perhaps we should look beyond just node count? Maybe the number of newline characters in the JSON?

Reply:
>  52MB un-minified, with node_attrs and all, and took 17 seconds to load for me

Due to the way this was uploaded (see [this slack thread for context](https://bedfordlab.slack.com/archives/C01LCTT7JNN/p1703021878712869)), it was not gzipped so that's going to make download performance / transfer size suffer a lot. (I would guess a lot more than new line characters.)

Reply:
I think 100 nodes is too few. If you look at measles (https://data.nextstrain.org/measles.json, https://nextstrain.org/measles) you'll see 260 tips and 471 nodes. Uncompressed and unminified it's 2.0 Mb. With gzip compression, it's 125 kb and took 162 ms to load.

I'd probably draw the line at about 1 Mb gzip compressed. 1000 nodes doesn't seem so far off to me.

Ah. If we look at mumps (https://data.nextstrain.org/mumps_na.json, https://nextstrain.org/mumps/na) we have 803 nodes. Uncompressed and unminified it's 2.1 Mb on disk (these branches have fewer mutations). So, yes. If using node metric I'd still suggest 1000.

I'd be okay with doing something like line endings instead as James suggests.

Reply:
FYI here are the stats on the file @victorlin mentioned, in decreasing order of file size:
* 52MB Starting JSON (uncompressed, unminimised)
* 4.8MB Minimised JSON (using `jq -c`)
* 1.4MB GZipped JSON (but _not_ minimised)
* 783KB Minimised + GZipped JSON

Reply:
If we assume the nextstrain CLI is the method datasets are uploaded, perhaps it could minimise every file as it uploads it? (It already gzips them)

Reply:
I don't think I'd minimize all files on upload. I like how I can look at look at and make sense of https://data.nextstrain.org/measles.json and it comes with what feels like minimal overhead for datasets of this size.

Reply:
Yeah fair enough. Gzip compression is so hidden away you've got to inspect the headers to know if it's even there, but viewing the raw JSON does change a lot with minimisation.

Reply:
Without minifying/compression, an important determinant of file is the indentation and level of nesting. we often have 100s of nesting levels, and if each is indented by a few characters, you get several kb of whitespace per line.

Reply:
Bikeshedding a bit, but an alternative to choosing a threshold number of nodes could be length of the json string, if it's >10^6 we could minify, that'd be the 1MB threshold and independent of how much of the JSON size comes from having many nodes, much nesting or many mutations.

Reply:
Thanks for the input everyone! Given the intent of minifying is to prevent long load times, which is often due to large file size, and that number of nodes alone is not a good indicator of file size, I think we need to use a different metric. I'll look into the alternatives mentioned above.

Reply:
Updated to use @corneliusroemer's JSON string length approach with a threshold of 5 MB. ba65ef239c5c03a7d21ac68b8a411653953a314a